weasyl/searchtag.py

```diff
@@ -118,6 +127,33 @@ def parse_tags(text):
     return tags
 
 
+def parse_blacklist_tags(text):
+    """
+    A custom implementation of ``parse_tags()`` for the searchtag blacklist.
+    Enforces the desired pattern of tag, and allows an asterisk character,
+       whereas ``parse_tags()`` would strip asterisks.
+
+    Parameters:
+        text: The string to parse for tags
+
+    Returns:
+        tags: A set() with valid tags.
+    """
+    tags = set()
+
+    for i in _TAG_DELIMITER.split(text):
+        target = "".join([c for c in i if ord(c) < 128])
+        target = target.replace(" ", "_")
+        target = "".join(i for i in target if i.isalnum() or i in "_*")
+        target = target.strip("_")
+        target = "_".join(i for i in target.split("_") if i)
+
+        if target.lower() and re.match(_SEARCHTAG_BLACKLIST_REGEXP_PATTERN, target):
```

Comment:
Checking `if target.lower()` doesn’t make much sense and is equivalent to `if target`.